src/lustre/lustreTypeChecker.ml

```diff
@@ -1425,6 +1429,10 @@ and check_type_well_formed: tc_context -> tc_type -> (unit, [> error]) result
             else Ok ()
       else type_error pos (SubrangeArgumentMustBeConstantInteger e2)
     else type_error pos (SubrangeArgumentMustBeConstantInteger e1)
+    | Some e1, None -> if is_expr_int_type ctx e1 then Ok () else type_error pos (SubrangeArgumentMustBeConstantInteger e1)
+    | None, Some e2 -> if is_expr_int_type ctx e2 then Ok () else type_error pos (SubrangeArgumentMustBeConstantInteger e2)
```

Comment:
The condition should also check `is_expr_of_consts ctx e1`. Similarly for `e2`.

Reply:
Addressed